### PR TITLE
doc: nRF Cloud REST lib: formatting fixes

### PR DESCRIPTION
**Pull request by peknis**
Fixed a Kconfig option misspelling and other
minor formatting issues.

Signed-off-by: Pekka Niskanen <pekka.niskanen@nordicsemi.no>

### CODE REVIEW

**Review by jayteemo — APPROVED** [review]

File: doc/nrf/libraries/networking/nrf_cloud_rest.rst [hunk 1]
```diff
@@ -31,24 +31,24 @@ Authentication
...
- * If you set ``keep_alive`` to true, the socket remains open and ``connect_socket`` remains unaltered, meaning the socket is reused on the next API call.
+* If you set ``keep_alive`` to false, the selected socket is closed after each request and the ``connect_socket`` property is reset to ``-1``.
+* If you set ``keep_alive`` to true, the socket remains open and ``connect_socket`` remains unaltered, meaning the socket is reused on the next API call.
 
-However, if you set ``keep_alive`` to true, make sure that the socket has not been closed externally (for example, due to inactivity) before sending each request.
+However, if you set ``keep_alive`` to true, make sure that the socket has not been closed externally (for example, due to inactivity) before sending a request.
 Otherwise, the request will be dropped.
 
-There are two ways a timed out socket can manifest:
+A timeout in a socket can manifest in two ways:
```
Comment (glarsennordic) [inline doc/nrf/libraries/networking/nrf_cloud_rest.rst:51]:
```suggestion
Timeouts in the socket can occur in two ways:
```

**Review by glarsennordic — APPROVED** [review]
LGTM, take suggestions under your own advisement

File: doc/nrf/libraries/networking/nrf_cloud_rest.rst [hunk 2]
```diff
@@ -31,24 +31,24 @@ Authentication
...
- * If you set ``keep_alive`` to true, the socket remains open and ``connect_socket`` remains unaltered, meaning the socket is reused on the next API call.
+* If you set ``keep_alive`` to false, the selected socket is closed after each request and the ``connect_socket`` property is reset to ``-1``.
+* If you set ``keep_alive`` to true, the socket remains open and ``connect_socket`` remains unaltered, meaning the socket is reused on the next API call.
 
-However, if you set ``keep_alive`` to true, make sure that the socket has not been closed externally (for example, due to inactivity) before sending each request.
+However, if you set ``keep_alive`` to true, make sure that the socket has not been closed externally (for example, due to inactivity) before sending further requests.
 Otherwise, the request will be dropped.
 
-There are two ways a timed out socket can manifest:
+Timeouts in the socket can occur in two ways:
```
Comment (greg-fer) [inline doc/nrf/libraries/networking/nrf_cloud_rest.rst:51]:
```suggestion
Timeouts in the socket can happen in two ways:
```
Less fancy word, but I always prefer simpler Saxon semantics than words of Latin origin.

**Review by greg-fer — APPROVED** [review]
Just one nit.

**Pull request merged**